server/src/test/java/org/elasticsearch/discovery/single/SingleNodeDiscoveryIT.java

```diff
@@ -66,6 +66,8 @@ protected Settings nodeSettings(int nodeOrdinal) {
     public void testSingleNodesDoNotDiscoverEachOther() throws IOException, InterruptedException {
         final TransportService service = internalCluster().getInstance(TransportService.class);
         final int port = service.boundAddress().publishAddress().getPort();
+        final int upperPortRangeBound = port + 5 - 1;
+        assumeTrue("port must be in range", upperPortRangeBound <= 65535);
```

Comment:
I was a bit going back and forth between adapting this test to use the JVM-local port range or just disabling the test in case it got unlucky with the randomly assigned port. I've gone with the prior solution now.